### PR TITLE
Ensure that plugins are sorted by name

### PR DESCRIPTION
**Pull request by res0nance**
Noticed this when looking at the code

### Proposed changelog entries

* Entry 1:  Available plugins should be sorted by name if popularity is equal

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/java/hudson/model/UpdateSite.java
Comment (daniel-beck) [inline core/src/main/java/hudson/model/UpdateSite.java:401]: oops 😅

**Review by daniel-beck — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
Hmm, I'm not sure if this code is actually used?

Available plugins page uses https://github.com/res0nance/jenkins/blob/updatesite-getAvailables/core/src/main/java/hudson/PluginManager.java#L1353

**Comment by timja** [comment]
on-holding till https://github.com/jenkinsci/jenkins/pull/5359#issuecomment-800049478 is clarified

**Comment by res0nance** [comment]
> Hmm, I'm not sure if this code is actually used?
> 
> Available plugins page uses https://github.com/res0nance/jenkins/blob/updatesite-getAvailables/core/src/main/java/hudson/PluginManager.java#L1353

Isn't it used just a couple lines below that?

https://github.com/res0nance/jenkins/blob/a244267356e7de3d3e2c3b23ce67392b6972df30/core/src/main/java/hudson/PluginManager.java#L1356

**Comment by timja** [comment]
It's used there but the previous sort order is ignored

**Comment by timja** [comment]
It's used there but the previous sort order is ignored

**Review by basil — APPROVED** [review]
I see no particular reason for this to be on hold assuming the new test fails without the changes to `core/src/main` and passes with them. A bug fix is a bug fix and stands on its own. Of course, if this code is generally redundant and/or unused, then it would be perfectly reasonable to [flense](https://en.wikipedia.org/wiki/Flensing) it in a separate PR.

**Comment by timja** [comment]
> I see no particular reason for this to be on hold assuming the new test fails without the changes to `core/src/main` and passes with them. A bug fix is a bug fix and stands on its own. Of course, if this code is generally redundant and/or unused, then it would be perfectly reasonable to [flense](https://en.wikipedia.org/wiki/Flensing) it in a separate PR.

it's more I'm not sure this is doing what the changelog says it's doing, i.e. it says we're fixing a bug, I'm not sure there is actually one there...

anyone we can ship it but maybe skip changelog?

**Comment by res0nance** [comment]
> > I see no particular reason for this to be on hold assuming the new test fails without the changes to `core/src/main` and passes with them. A bug fix is a bug fix and stands on its own. Of course, if this code is generally redundant and/or unused, then it would be perfectly reasonable to [flense](https://en.wikipedia.org/wiki/Flensing) it in a separate PR.
> 
> it's more I'm not sure this is doing what the changelog says it's doing, i.e. it says we're fixing a bug, I'm not sure there is actually one there...
> 
> anyone we can ship it but maybe skip changelog?

Sure, not sure if this affects anything right now, the function does have the `@Exported` annotation so it might

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
